Update getHandlerPath.ts

yehonadav opened the pull request:
make it cross-platform

fargito reviewed (approved):
Thanks a lot for this!

fargito commented:
It seems there is a linter error, have you run prettier on the file?

fargito commented:
Ok, thanks for investigating, I'm going to set this as draft then

yehonadav commented:
just updating here, when i revert the changes i made, serverless fails to package with:
```
    ×  nx run backend-core:build
       > @swarmion-full-stack/backend-core@1.0.0 build C:\git\yb\swarmion-full-stack\backend\core
> serverless package


Packaging swarmion-full-stack-core for stage dev (eu-west-1)
Cannot locate handler - C:\git\yb\swarmion-full-stack\backend\core\functions\health/handler not found
Environment: win32, node 16.15.0, framework 3.23.0 (local), plugin 6.2.2, SDK 4.3.2
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
Error: Compilation failed. Please ensure you have an index file with ext .ts or .js, or have a path listed as main key in package.json
    at C:\git\yb\swarmion-full-stack\node_modules\.pnpm\serverless-esbuild@1.33.0_esbuild@0.14.54\node_modules\serverless-esbuild\dist\helper.js:57:15
    at Array.map (<anonymous>)
    at extractFunctionEntries (C:\git\yb\swarmion-full-stack\node_modules\.pnpm\serverless-esbuild@1.33.0_esbuild@0.14.54\node_modules\serverless-esbuild\dist\helper.js:36:35)
    at EsbuildServerlessPlugin.get functionEntries [as functionEntries] (C:\git\yb\swarmion-full-stack\node_modules\.pnpm\serverless-esbuild@1.33.0_esbuild@0.14.54\node_modules\serverless-esbuild\dist\index.js:209:52)
    at EsbuildServerlessPlugin.bundle (C:\git\yb\swarmion-full-stack\node_modules\.pnpm\serverless-esbuild@1.33.0_esbuild@0.14.54\node_modules\serverless-esbuild\dist\bundle.js:81:48)
    at before:package:createDeploymentArtifacts (C:\git\yb\swarmion-full-stack\node_modules\.pnpm\serverless-esbuild@1.33.0_esbuild@0.14.54\node_modules\serverless-esbuild\dist\index.js:67:28)
    at PluginManager.runHooks (C:\git\yb\swarmion-full-stack\node_modules\.pnpm\serverless@3.23.0\node_modules\serverless\lib\classes\plugin-manager.js:530:15)
    at PluginManager.invoke (C:\git\yb\swarmion-full-stack\node_modules\.pnpm\serverless@3.23.0\node_modules\serverless\lib\classes\plugin-manager.js:563:20)
    at async PluginManager.run (C:\git\yb\swarmion-full-stack\node_modules\.pnpm\serverless@3.23.0\node_modules\serverless\lib\classes\plugin-manager.js:604:7)
    at async Serverless.run (C:\git\yb\swarmion-full-stack\node_modules\.pnpm\serverless@3.23.0\node_modules\serverless\lib\serverless.js:174:5)
    at async C:\git\yb\swarmion-full-stack\node_modules\.pnpm\serverless@3.23.0\node_modules\serverless\scripts\serverless.js:771:9
 ELIFECYCLE  Command failed with exit code 1.
```

so the previous solution made the handler path looks like this:
C:\git\yb\swarmion-full-stack\backend\core\functions\health\handler.main

but what we actually want is
functions/health/handler.main

so maybe something like this is a solution to consider
```ts
import * as path from 'path';

/** Helper to be used in config.ts files to retrieve the path of the handler. */
const getHandlerPath = (directoryPath: string): string => {
  const processRunLocation = process.cwd();

  const handlerPath = directoryPath.replace(processRunLocation + path.sep, '') + path.sep + 'handler.main';

  return 'functions' + handlerPath.split('functions', 2)[1].split(path.sep).join('/');
};

export default getHandlerPath;
```

yehonadav commented:
@fargito i made the suggested changes and am leaving it to your consideration

Eszik commented:
Hello, any chance this could get merged? I'm currently running into some issues using the getHandlerPath method on Windows
